addons/account_edi/models/account_edi_document.py

```diff
@@ -0,0 +1,177 @@
+            if doc_type == 'payment':
+                payments.append((key, batches))
+                continue  # payments are processed after invoices
+
+            for documents in batches:
+                try:
+                    with self.env.cr.savepoint():
+                        # Locks the documents in DB. Avoid sending an invoice twice (the documents can be processed by the CRON but also manually).
+                        self._cr.execute('SELECT * FROM account_edi_document WHERE id IN %s FOR UPDATE NOWAIT', [tuple(self.ids)])
+
+                        if state == 'to_send':
+                            edi_result = edi_format._post_invoice_edi(documents.move_id, test_mode=test_mode)
+                            _postprocess_post_edi_results(documents, edi_result)
+                        elif state == 'to_cancel':
+                            edi_result = edi_format._cancel_invoice_edi(documents.move_id, test_mode=test_mode)
+                            _postprocess_cancel_edi_results(documents, edi_result)
+
+                except OperationalError as e:
+                    if e.pgcode == '55P03':
+                        _logger.debug('Another transaction already locked documents rows. Cannot process documents.')
+                    else:
+                        raise e
+
+        # ==== Process payments ====
+        for key, batches in payments:
+            edi_format, state, doc_type, company_id = key
+
+            for documents in batches:
+                try:
+                    with self.env.cr.savepoint():
+                        self._cr.execute('SELECT * FROM account_edi_document WHERE id IN %s FOR UPDATE NOWAIT', [tuple(self.ids)])
+
+                        if state == 'to_send':
+                            edi_result = edi_format._post_payment_edi(documents.move_id, test_mode=test_mode)
+                            _postprocess_post_edi_results(documents, edi_result)
+                        elif state == 'to_cancel':
+                            edi_result = edi_format._cancel_payment_edi(documents.move_id, test_mode=test_mode)
+                            _postprocess_cancel_edi_results(documents, edi_result)
+
+                except OperationalError as e:
+                    if e.pgcode == '55P03':
+                        _logger.debug('Another transaction already locked documents rows. Cannot process documents.')
+                    else:
+                        raise e
+
+    def _process_documents_no_web_services(self):
+        """ Post and cancel all the documents that don't need a web-service.
+        """
+        jobs = self.filtered(lambda d: not d.edi_format_id._needs_web_services())._prepare_jobs()
+        self._process_jobs(jobs)
+
+    def _process_documents_web_services(self, job_count=None):
+        """ Post and cancel all the documents that need a web-service. This is called by CRON.
+
+        :param job_count: Limit the number of documents that can be to post or cancel.
```

Comment:
This sentence is weid. 
"Limit to the number of jobs to process among the ones that are available for treatment" ?